### ISSUE TITLE
False positive: class passed to function expecting `Protocol[ParamSpec]` not recognized as structural match

### ISSUE THREAD

**Issue by jjovalle99** [astral-sh/ty#2930]
### Summary

ty emits `invalid-argument-type` when passing a class to a function that expects a `Protocol[ParamSpec]`, even though the class structurally satisfies the protocol. Both pyright, mypy, and runtime execution accept the code.

**Minimal reproducer:**

```python
from collections.abc import Awaitable, Callable
from typing import Any, ParamSpec, Protocol

P = ParamSpec("P")

_ASGIApp = Callable[[Any, Any, Any], Awaitable[None]]


class _MiddlewareFactory(Protocol[P]):
    def __call__(self, app: _ASGIApp, /, *args: P.args, **kwargs: P.kwargs) -> _ASGIApp: ...


def add_middleware(middleware_class: _MiddlewareFactory[P], *args: P.args, **kwargs: P.kwargs) -> None: ...


class MyMiddleware:
    def __init__(self, app: _ASGIApp, /, *, setting: str = "") -> None:
        self.app = app
        self.setting = setting

    async def __call__(self, scope: Any, receive: Any, send: Any) -> None:
        await self.app(scope, receive, send)


add_middleware(MyMiddleware, setting="value")
```

**ty output:**

```
error[invalid-argument-type]: Argument to function `add_middleware` is incorrect
  --> repro.py:25:16
   |
25 | add_middleware(MyMiddleware, setting="value")
   |                ^^^^^^^^^^^^ Expected `_MiddlewareFactory[(...)]`, found `<class 'MyMiddleware'>`
```

**Expected:** No error. `MyMiddleware` structurally satisfies `_MiddlewareFactory` — calling `MyMiddleware(app, *, setting="value")` returns a `MyMiddleware` instance, which is callable with `(scope, receive, send) -> Awaitable[None]`, satisfying `_ASGIApp`.

**pyright 1.1.398:** 0 errors, 0 warnings.
**mypy 1.15.0:** Success, no issues found.

**Real-world impact:** This pattern is used by Starlette/FastAPI's `add_middleware()`, which uses the exact `_MiddlewareFactory` protocol above (defined in `starlette.middleware`). Every middleware class (including `CORSMiddleware`, `TrustedHostMiddleware`, `GZipMiddleware`, etc.) triggers this false positive, forcing `# ty: ignore[invalid-argument-type]` on every `add_middleware()` call.

Related upstream discussions:
- https://github.com/encode/starlette/discussions/2423
- https://github.com/Kludex/starlette/discussions/2757
- https://github.com/Kludex/starlette/pull/3059

Possibly related to #2926 (ParamSpec + Protocol interaction).

### Version

ty 0.0.19 (ae10022c2 2026-02-26)
